default-token-account/program/src/processor.rs

```diff
@@ -0,0 +1,167 @@
+//! Program state processor
+
+#![cfg(feature = "program")]
+
+use crate::{
+    create_default_token_account_address, error::DefaultTokenAccountError,
+    instruction::DefaultTokenAccountInstruction,
+};
+use num_traits::FromPrimitive;
+#[cfg(target_arch = "bpf")]
+use solana_sdk::program::{invoke, invoke_signed};
+use solana_sdk::{
+    account_info::{next_account_info, AccountInfo},
+    decode_error::DecodeError,
+    entrypoint::ProgramResult,
+    info,
+    program_error::PrintProgramError,
+    program_pack::Pack,
+    pubkey::Pubkey,
+    system_instruction,
+};
+
+use spl_token::{self, state::Account};
+
+/// Program state handler.
+pub struct Processor {}
+impl Processor {
+    /// Process a DefaultTokenAccountInstruction::Create instruction
+    pub fn process_create(
+        _program_id: &Pubkey,
+        bump_seed: u8,
+        accounts: &[AccountInfo],
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+
+        let default_token_account_info = next_account_info(account_info_iter)?;
+        let user_wallet_info = next_account_info(account_info_iter)?;
+        let mint_info = next_account_info(account_info_iter)?;
+        let sysvar_rent = next_account_info(account_info_iter)?;
+
+        let default_token_account_signer_seeds: &[&[_]] = &[
+            &user_wallet_info.key.to_bytes(),
+            &mint_info.key.to_bytes(),
+            &[bump_seed],
+        ];
+
+        // Allocate space for the default token account
+        invoke_signed(
+            &system_instruction::allocate(default_token_account_info.key, Account::LEN as u64),
+            &[default_token_account_info.clone()],
+            &[default_token_account_signer_seeds],
+        )?;
+
+        // Assign the default token account to the SPL Token program
+        invoke_signed(
+            &system_instruction::assign(default_token_account_info.key, &spl_token::id()),
+            &[default_token_account_info.clone()],
+            &[default_token_account_signer_seeds],
+        )?;
+
+        // Initialize the default token account
+        invoke_signed(
```

Comment:
I think you can just use `invoke()` here

Reply:
Yep.  I didn't at the time because then I'd need to stub out `invoke()` like `invoke_signed()`.   But I'm knee deep into overhauling the non-BPF syscalls in `sdk/src` and with my latest local code, moving back to `invoke()` just works!